code/java/index1.java

```diff
@@ -0,0 +1,103 @@
+package code.java;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.xapian.Document;
+import org.xapian.Stem;
+import org.xapian.TermGenerator;
+import org.xapian.WritableDatabase;
+import org.xapian.XapianConstants;
+import org.xapian.XapianJNI;
+
+public class index1 {
+    
+    // Command line args - dbpath datapath
+    public static void main(String[] args)
+    {
+        if(args.length < 2)
+        {
+            System.out.println("Insufficient number of arguments (should be dbpath datapath)");
+            return;
+        }
+        index(args[0], args[1]);
+    }
+    
+    public static void index(String dbpath, String datapath)
+    {
+        // Create or open the database we're goign to be writing to.
+        WritableDatabase db = new WritableDatabase(dbpath, XapianConstants.DB_CREATE_OR_OPEN);
+        // Set up a TermGenerator that we'll use in indexing.
+        TermGenerator termGenerator = new TermGenerator();
+        termGenerator.setStemmer(new Stem("en"));
+        
+        //Parsing the CSV input file
+        Scanner csvScanner,lineScanner;        
+        csvScanner = lineScanner = null;
+        
+        try {
+            File csv = new File(datapath);
+            csvScanner = new Scanner(csv);
+        } catch (FileNotFoundException ex) {
+        	Logger.getLogger(index1.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        
+        // Ignoring first line (contains descriptors)
+        csvScanner.nextLine();
+
+        while(csvScanner.hasNextLine())
+        {
+            String currentLine = csvScanner.nextLine();
+            lineScanner = new Scanner(currentLine);
+            lineScanner.useDelimiter(",");
```

Comment:
This won't parse CSV files properly, because it doesn't cope with commas in values (where the entire value is then quoted). In C++ we had to write our own parsing library, and that may be what's required here.